cogito/protocol.go

```diff
@@ -257,7 +261,12 @@ func (src *Source) Validate() error {
 	//
 	// Validate optional fields.
 	//
-	// In this case, nothing to validate.
+	if src.GithubApiEndpoint != "" {
+		u, err := url.ParseRequestURI(src.GithubApiEndpoint)
+		if err != nil || u.Host == "" {
+			return fmt.Errorf("source: github_api_endpoint '%s' is an invalid api endpoint", src.GithubApiEndpoint)
```

Comment:
Here I have the impression that we are removing details from the error that we actually have and would help the user, or do we have a particular reason?

```go
if src.GithubApiEndpoint != "" {
    u, err := url.ParseRequestURI(src.GithubApiEndpoint)
    if err != nil {
        return fmt.Errorf("source: invalid github_api_endpoint %q: %s", src.GithubApiEndpoint, err)
    }
    if u.Host == "" {
        return fmt.Errorf("source: invalid github_api_endpoint %q: %s", src.GithubApiEndpoint, "empty hostname")
    }
}
```

Reply:
Changed as suggested.